kani-compiler/src/kani_middle/mod.rs

```diff
@@ -127,20 +127,16 @@ pub fn dump_mir_items(tcx: TyCtxt, items: &[MonoItem], output: &Path) {
 pub struct SourceLocation {
     pub filename: String,
     pub start_line: usize,
-    pub start_col: usize,
     pub end_line: usize,
-    pub end_col: usize,
```

Comment:
Assuming you're getting warnings from the linter here, do they still happen if you remove the `pub` keyword?

Reply:
Why would removing `pub` be the right fix when these are unused members? (Removal was done in response to https://github.com/model-checking/kani/pull/3116#discussion_r1544682573.)

Reply:
I'm trying to understand what's the problem first. Does `clippy` throw an error because you don't read the value ever? Or because you don't read the value AND it's public?

The removal of these fields in `struct`s that derive `Serialize` or `Deserialize` would cause (de)serialization to stop working altogether. That's not the case here, but it's what prevents you from removing the attributes in `LitaniRun`, for example. Moreover, both you and I know that a source location contains columns, and removing it here might be confusing for other developers who need to use them later.

I think this warning could be silenced by either removing `pub` or adding an underscore `_` at the beginning of the field names. Again, I don't even know what the warning is, I just want us to be reasonable when it comes to `clippy` warnings and such.

Reply:
`rustc` yields a warning, and `clippy` makes that (in our configuration) an error. I have now added `#[allow(dead_code)]` for just those two fields and also added a comment.